hcp/hcl_handler.go

```diff
@@ -23,105 +22,114 @@ const (
 	buildLabel                 string = "build"
 )
 
-// TrySetupHCP attempts to setup the HCP-related structures if HCP is enabled
-// for the command
-func TrySetupHCP(cfg packer.Handler) hcl.Diagnostics {
-	switch cfg := cfg.(type) {
-	case *hcl2template.PackerConfig:
-		return setupRegistryForPackerConfig(cfg)
-	case *CoreWrapper:
-		return setupRegistryForPackerCore(cfg)
-	}
+type HCLHandler struct {
```

Comment:
Guessing this is meant to be HCPHandler and not HCLHandler. Which is the same name as the HCPHandler interface so that wont work. Let think about this a bit more to see what to call it. We probably want to rename the HCPHandler interface as well.

Reply:
In this case it's intended to be `HCLHandler` since it handles HCL templates, but the name is confusing I agree, maybe we can see to replace it by something else.
I should probably add some comment on top to clarify why it's here though, the whole package is poorly documented for now, so I'll amend that for my next reroll.